include/AudioDevice.h

```diff
@@ -75,66 +75,52 @@ class AudioDevice
 		return m_sampleRate;
 	}
 
-	void processNextBuffer();
+	fpp_t framesPerPeriod() const { return m_framesPerPeriod; }
 
-	virtual void startProcessing()
-	{
-		m_inProcess = true;
-	}
-
-	virtual void stopProcessing();
+	virtual void startProcessing() { m_running.test_and_set(std::memory_order_acquire); }
+	virtual void stopProcessing() { m_running.clear(std::memory_order_release); }
```

Comment:
Instead of needing to remember to call `AudioDevice::startProcessing` and `AudioDevice::stopProcessing` from the overridden methods in the child classes, why not let these be non-virtual wrappers that call some new pure virtual methods that the child classes would implement?

For example:
```cpp
public:
	void startProcessing()
	{
		m_running.test_and_set(std::memory_order_acquire);
		startProcessingImpl();
	}

protected:
	virtual void startProcessingImpl() = 0;
```

Reply:
I like this idea.

Reply:
Though, I think `startProcessingImpl` can actually be made `private` here.